cachi2/core/package_managers/pip.py

```diff
@@ -1300,11 +1301,8 @@ def from_line(cls, line: str, options: list[str]) -> Optional["PipRequirement"]:
             requirement.kind = "pypi"
 
         try:
-            parsed: Sequence[Requirement] = list(pkg_resources.parse_requirements(to_be_parsed))
-        except (
-            RequirementParseError,
-            pkg_resources.extern.packaging.requirements.InvalidRequirement,
-        ) as exc:
+            parsed: Sequence[Requirement] = list(_parse_requirements(to_be_parsed))
```

Comment:
Doing this effectively just vendors the old pkg_resources code same way as pkg_resources vendors packaging from which it builds upon. IOW you can drop the whole `_parse_requirements` parsing into a list and instead do plain `packaging.Requirement(to_be_parsed)`. Note that tests will have to be adjusted since we're checking some corner cases that relate to `_parse_requirements` behaviour which no longer apply to the usage of `packaging.Requirement`.